tests/integration/tests/charm_tests/openstack_exporter.py

```diff
@@ -256,18 +257,45 @@ def test_openstack_exporter_snap_down(self):
         )
         self.assertEqual(self.leader_unit.workload_status_message, "")
 
+    def test_upstream_exporter_as_resource(self):
+        """Test using the upstream golang as resource will block the unit."""
+        # juju cannot access resources at /tmp, so we create a tmp folder at the current directory
+        with tempfile.TemporaryDirectory(dir="./") as tmpdir:
+            tmp_path = Path(tmpdir)
+            download_cmd = (
+                f"wget -q -P {tmpdir} https://github.com/canonical/openstack-exporter-operator/"
+                "releases/download/rev2/golang-openstack-exporter_amd64.snap"
+            )
+            subprocess.run(download_cmd.split(), check=False)
```

Comment:
I'm not sure about this... I don't think we can rely on an old release snap download url.  I guess it's fine to keep for now; eventually we can drop the logic that manages the old snap altogether after a certain deprecation period.

Reply:
Yes, I agree that we can possibly remove this logic after some time.